ql/src/java/org/apache/hadoop/hive/ql/io/orc/VectorizedOrcAcidRowBatchReader.java

```diff
@@ -959,6 +989,20 @@ public boolean next(NullWritable key, VectorizedRowBatch value) throws IOExcepti
       int ix = rbCtx.findVirtualColumnNum(VirtualColumn.ROWID);
       value.cols[ix] = recordIdColumnVector;
     }
+    if (rowIsDeletedProjected) {
+      if (fetchDeletedRows) {
```

Comment:
if (!fetchDeletedRows || notDeletedBitSet.cardinality() == vectorizedRowBatchBase.size )

Reply:
tbh we could even do the second check as part of the Set method (as we do already for cardinality 0) and simplify the logic here

Reply:
I prefer your first suggestion because the second one requires passing `vectorizedRowBatchBase.size()` to the `set` method which I would like to avoid.